src/java/org/apache/cassandra/repair/SystemDistributedKeyspace.java

```diff
@@ -307,6 +320,55 @@ public static void setViewRemoved(String keyspaceName, String viewName)
         forceBlockingFlush(VIEW_BUILD_STATUS);
     }
 
+    /**
+     * Reads blacklisted partitions from system_distributed.blacklisted_partitions table.
+     * Stops reading partitions upon exceeding the cache size limit by logging a warning.
+     * @return
+     */
+    public static Set<BlacklistedPartition> getBlacklistedPartitions()
+    {
+        String query = "SELECT keyspace_name, columnfamily_name, partition_key FROM %s.%s";
+        UntypedResultSet results;
+        try
+        {
+            results = QueryProcessor.execute(format(query, SchemaConstants.DISTRIBUTED_KEYSPACE_NAME, BLACKLISTED_PARTITIONS),
```

Comment:
Just double checking, does this paginate?

Reply:
based on my quick glance, I don't think it does. Even if it does paginate, not sure how we can avoid possible OOM (without inspecting result size after each paginated query).

Reply:
I think there is a local execution with pagination option (which should avoid oom right)?

Reply:
execute() call can be replaced with executeInternalWithPaging(), but with paging or otherwise, we will still end up holding entire result set in memory through "results" variable here, isn't it? You think it could still help using paging?